tests/dtm/test_single_process_restart.py

```diff
@@ -150,24 +177,167 @@ def test_write_during_m0d_restart(self):
+        output = Queue()
+
+        self.log.info("Create IAM user with name %s", self.s3acc_name)
+        resp = self.rest_obj.create_s3_account(acc_name=self.s3acc_name,
+                                               email_id=self.s3acc_email,
+                                               passwd=S3_CFG["CliConfig"]["s3_account"]["password"])
+        assert_utils.assert_true(resp[0], resp[1])
+        access_key = resp[1]["access_key"]
+        secret_key = resp[1]["secret_key"]
+        self.iam_user = {'s3_acc': {'accesskey': access_key, 'secretkey': secret_key,
+                                    'user_name': self.s3acc_name}}
+        test_prefix = 'test-41234'
+        self.log.info("Create buckets for IOs")
+        s3_test_obj = S3TestLib(access_key=access_key, secret_key=secret_key,
+                                endpoint_url=S3_CFG["s3_url"])
+        workloads = HA_CFG["s3_bench_workloads"]
+        if self.setup_type == "HW":
+            workloads.extend(HA_CFG["s3_bench_large_workloads"])
+        for workload in workloads:
+            s3_test_obj.create_bucket(f"bucket-{workload.lower()}-{test_prefix}")
+
+        self.log.info("Perform WRITEs/READs-Verify with variable object sizes in background")
+        args = {'s3userinfo': self.iam_user, 'log_prefix': test_prefix,
+                'nclients': test_cfg['nclients'], 'nsamples': test_cfg['nsamples'],
+                'skipcleanup': True, 'output': output}
+
+        thread = threading.Thread(target=self.ha_obj.event_s3_operation,
+                                  args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+
+        self.log.info("Step 1: Successfully created IAM user and started WRITEs/READs-Verify "
+                      "with variable object sizes in background")
+        self.log.info("Sleep for %s sec", HA_CFG["common_params"]["30sec_delay"])
+        time.sleep(HA_CFG["common_params"]["30sec_delay"])
+
+        self.log.info("Step 2: Perform single restart of m0d process on pod hosted on RC node and "
+                      "check hctl status")
+        self.log.info("Get RC node name")
+        rc_node = self.ha_obj.get_rc_node(self.master_node_list[0])
+        rc_info = self.master_node_list[0].get_pods_node_fqdn(pod_prefix=rc_node.split("svc-")[1])
+        rc_node_name = list(rc_info.values())[0]
+        self.log.info("RC Node is running on %s node", rc_node_name)
+        self.log.info("Get the data pod running on %s node", rc_node_name)
+        data_pods = self.master_node_list[0].get_pods_node_fqdn(const.POD_NAME_PREFIX)
+        rc_datapod = None
+        for pod_name, node in data_pods.items():
+            if node == rc_node_name:
+                rc_datapod = pod_name
+                break
+        self.log.info("RC node %s has data pod: %s ", rc_node_name, rc_datapod)
+        event.set()
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
```

Comment:
Is just checking for failure enough. If the redo logs comes up incorrect then?  Can you check for consistency of your write ?

Reply:
Fix this at all the places.

Reply:
@seagate-sarang-sawant We are performing the read/validate operations after the recovery is done,
If redo logs are incorrect, read/validate operations may fail. 
Also, redo logs are metadata logs and are not easily readable for us. 
These are functional tests for dtm and hence explicit DI check can be done in DI testing.

Reply:
No we want consistency to be check in the consistency test cases itself.  The aim of DI will not be this.

Reply:
If redo logs are incorrect, object entry consistency will go for a toss. Checksum verification of the data stored just before trigger of m0 recovery at run time may detect that.

Reply:
@seagate-sarang-sawant I think we are doing it here. 
We also have dedicated tests for writes, reads, deletes during m0d restart.

Reply:
@seagate-sarang-sawant we can resolve as per our discussion